File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        Queue<Integer> queue = new ArrayDeque<>();
        
        for(int i = 1; i <= n; i++) {
            queue.add(i);
        }
        
        while(queue.size() > 1) {
            // System.out.println(queue);
            for(int i = 1; i < k; i++) {
                queue.add(queue.remove());
            }
            queue.remove();
        }
        
        return queue.remove();
    }
}

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        queue = list()
        for i in range(1, n + 1):
            queue.append(i)
        
        while len(queue) > 1:                
            # print(queue)
            for i in range(k - 1):
                queue.append(queue[0])
                del queue[0]
                
            del queue[0]
            
        return queue[0]